src/coreclr/jit/lowerxarch.cpp

```diff
@@ -1517,26 +1517,24 @@ void Lowering::LowerHWIntrinsicCreate(GenTreeHWIntrinsic* node)
 
         assert((simdSize == 8) || (simdSize == 12) || (simdSize == 16) || (simdSize == 32));
 
-        if ((argCnt == 1) ||
-            ((vecCns.i64[0] == vecCns.i64[1]) && ((simdSize <= 16) || (vecCns.i64[2] == vecCns.i64[3]))))
+        if (((simdSize == 16) || (simdSize == 32)) &&
+            VectorConstantIsBroadcastedI64(vecCns, (argCnt == 1) ? argCnt : (simdSize / 8)))
```

Comment:
same question, if `argCnt == 1` does not it mean `VectorConstantIsBroadcastedI64(vecCns, simdSize / 8)` returns true?

Reply:
I've just checked - yeah we can omit argCnt==1 logic for simplicity here, thanks